### PR TITLE
Rel-5_0 - OTRS - updated unit test - part 6

### PR DESCRIPTION
**Pull request by s7design**
Hi @mgruner 

There is one more PR with updated unit test where are added Helper object with params for cleaning-up system after running tests. I also fixed some other test code, I hope I improved it but if you have any suggestion about that please let me know.

Regards
Zoran

### CODE REVIEW

File: scripts/test/ReferenceData.t [hunk 1]
```diff
@@ -14,9 +14,17 @@ use vars (qw($Self));
 
 use Kernel::System::VariableCheck qw(:all);
 
-# get needed objects
+# get config object
 my $ConfigObject = $Kernel::OM->Get('Kernel::Config');
 
+# get helper object
+$Kernel::OM->ObjectParamAdd(
```
Comment (mgruner) [inline scripts/test/ReferenceData.t]: This seems unneeded as there are no SysConfig calls here, only ConfigObject that changes only runtime?

File: scripts/test/MailAccount.t [hunk 2]
```diff
@@ -12,9 +12,7 @@ use utf8;
 
 use vars (qw($Self));
 
-use Kernel::System::ObjectManager;
-
-# get needed objects
+# get mail account object
```
Comment (mgruner) [inline scripts/test/MailAccount.t]: No RestoreDatabase here?

File: scripts/test/Loader.t [hunk 3]
```diff
@@ -14,21 +14,28 @@ use vars (qw($Self));
...
 # get needed objects
 my $ConfigObject = $Kernel::OM->Get('Kernel::Config');
-my $HelperObject = $Kernel::OM->Get('Kernel::System::UnitTest::Helper');
 my $LoaderObject = $Kernel::OM->Get('Kernel::System::Loader');
 my $MainObject   = $Kernel::OM->Get('Kernel::System::Main');
 
+# get helper object
+$Kernel::OM->ObjectParamAdd(
+    'Kernel::System::UnitTest::Helper' => {
+        RestoreDatabase => 1,
```
Comment (mgruner) [inline scripts/test/Loader.t]: Do we really need this here? I believe the Loader does not change the DB.

**Comment by mgruner** [comment]
Hello Zoran, please see my inline comments.

File: scripts/test/MailAccount.t (hunk 2 above)
Comment (s7design) [inline scripts/test/MailAccount.t]: It is actually not need here, there is MailAccountDelete at the end of the test.

File: scripts/test/ReferenceData.t (hunk 1 above)
Comment (s7design) [inline scripts/test/ReferenceData.t]: Thanks, I will fix it, I added RestoreDtaabase, but I did not think that it is only ConfigObject.

File: scripts/test/Loader.t (hunk 3 above)
Comment (s7design) [inline scripts/test/Loader.t]: I addded it only to clear cache, I will change comment at the end in:
# cleanup cache is done by RestoreDatabase

**Comment by s7design** [comment]
The PR is updated. Please let me know if I can fix anything else.

Regards
Zoran

**Comment by mgruner** [comment]
Merged, thanks. I still added RestoreDatabase to MailAccount.t as it avoids write operations in the DB and we can really make sure there are no leftovers in any case, also for future additions to the test.

**Pull request closed without merging**